project/Build.scala

```diff
@@ -319,10 +319,11 @@ object Build {
 
   private lazy val currentYear: String = java.util.Calendar.getInstance().get(java.util.Calendar.YEAR).toString
 
-  lazy val scalacOptionsDocSettings = Seq(
-      "-external-mappings:" +
-        ".*scala/.*::scaladoc3::https://dotty.epfl.ch/api/," +
-        ".*java/.*::javadoc::https://docs.oracle.com/javase/8/docs/api/",
+  def scalacOptionsDocSettings(includeExternalMappings: Boolean = true) = {
```

Comment:
Is this part needed for this fix?

Reply:
Yes, because by default we were adding a external mapping to dotty.epfl.ch in Scala documentation. Our mechanism to resolve links tries to match external mapping for symbols that are not present in docs. Since the private API was not present in docs, Scaladoc matched external mapping for it and generated unexisting link

Reply:
ok, makes sense

Reply:
Isn't it a general problem? I mean, maybe we should not try to match any link for members that are private, not workaround it this way

Reply:
That's why I introduced changes with filtering symbols that shouldn't be visible. Maybe we should once more go through whole code that gathers symbols and check if we filter all symbols correctly. 

I removed the external link to dotty because it was redundant and should never be there.